In [1]:
from keras.models import Sequential
from keras.layers import Conv2D

model = Sequential()
model.add(Conv2D(filters=16, kernel_size=2, strides=2, padding='valid', activation='relu', input_shape=(200, 200, 1)))
model.summary()

C:\Python36\lib\site-packages\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.


_________________________________________________________________
Layer (type)                 Output Shape              Param #   
conv2d_1 (Conv2D)            (None, 100, 100, 16)      80        
Total params: 80
Trainable params: 80
Non-trainable params: 0
_________________________________________________________________


#### Note:

- Number of param = (kernel_size x kernel_size x depth x filters) + filters = (2 x 2 x 1 x 16) +16 = 80
- The ***depth*** of the convolutional layer will always = filters
- If ***padding = 'same'*** == ADD PADDING :
    - height = ceil(float(Height of img in prev layer) / float(stride)) = ceil(200/2) = 100
    - width = ceil(float(Width of img in prev layer) / float(stride)) = ceil(200/2) = 100
- If ***padding = 'valid'*** == IGNORE edges:
    - height = ceil(float(Height of img in prev layer - kernel_size + 1) / float(stride))
    - width = ceil(float(Width of img in prev layer - kernel_size + 1) / float(stride))

In [3]:
#Another Example
model = Sequential()
model.add(Conv2D(filters=32, kernel_size=3, strides=2, padding='same', activation='relu', input_shape=(128, 128, 3)))
model.summary()

_________________________________________________________________
Layer (type)                 Output Shape              Param #   
conv2d_2 (Conv2D)            (None, 64, 64, 32)        896       
Total params: 896
Trainable params: 896
Non-trainable params: 0
_________________________________________________________________


Here,
- Number of params = (3 x 3 x 3 x 32) + 32 = 896
- Depth = 32
- Width of convolutional layer = ceil(128/kernel_size=3) = 64